pkg/querier/batch/merge_test.go

```diff
@@ -61,3 +63,47 @@ func TestMergeHarder(t *testing.T) {
 		})
 	}
 }
+
+// TestBugInMergeIterator tests a bug while calling Seek() on mergeIterator.
+func TestBugInMergeIteratorSeek(t *testing.T) {
+	// Samples for 3 chunks.
+	chunkSamples := [][]int64{
+		{10, 20, 30, 40},
+		{50, 60, 70, 80, 90, 100},
+		{110, 120},
+	}
+
+	var genericChunks []GenericChunk
+	for _, samples := range chunkSamples {
+		ch := chunkenc.NewXORChunk()
+		app, err := ch.Appender()
+		require.NoError(t, err)
+		for _, ts := range samples {
+			app.Append(ts, 1)
+		}
+
+		genericChunks = append(genericChunks, NewGenericChunk(samples[0], samples[len(samples)-1], func(reuse chunk.Iterator) chunk.Iterator {
+			chk, err := chunk.NewForEncoding(chunk.PrometheusXorChunk)
+			require.NoError(t, err)
+			require.NoError(t, chk.UnmarshalFromBuf(ch.Bytes()))
+			return chk.NewIterator(reuse)
+		}))
+	}
+
+	c3It := NewGenericChunkMergeIterator(nil, labels.EmptyLabels(), genericChunks)
+
+	c3It.Seek(15)
+	// These Next() calls are necessary to reproduce the bug.
```

Comment:
are they necessary so that we cross the chunk boundary or?

Reply:
Yes I think so. These three Next() cross from chunk 1 to 2. I will probably simply the test if it still reproduces the earlier bug

Reply:
Looks like there is something specific with these three Next() that reproduces it with this input. I don't think I want to dig further into details of why at the moment :)